openverse_catalog/dags/providers/provider_api_scripts/smithsonian.py

```diff
@@ -202,16 +202,33 @@ def get_new_and_outdated_unit_codes(
     return new_unit_codes, outdated_unit_codes
 
 
+def get_unit_codes_from_api(
```

Comment:
We added the [`retry` library](https://pypi.org/project/retry/) a while back in #330 to address similar flakiness with the Freesound provider script. It might be worth using the `@retry` decorator and raising a specific error if the list returned is empty. That way we 1) put some automatic delay in between our requests and 2) raise an error if we _still_ get an empty list after several retries. I think the current implementation would pass the empty list if 3 retries resulted in an empty response, and leave us with the same slack notifications we're receiving now 😅